ci/environment.yml

```
@@ -2,26 +2,21 @@ build & push image:
   stage: setup environment
   when: manual
   before_script:
+    - docker login gitlab.com/satoshilabs/dependency_proxy/containers/ -u $CI_DEPENDENCY_PROXY_USER -p $CI_DEPENDENCY_PROXY_PASSWORD
     - docker login $CI_REGISTRY -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD
   variables:
     DOCKER_TLS_CERTDIR: ''
     CONTAINER_NAME: "$CI_REGISTRY/satoshilabs/trezor/trezor-suite/base"
-  image: docker
+  image: $CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX/docker
   services:
     - docker:dind
   script:
     - docker pull $CONTAINER_NAME:latest || true
-    - docker build --no-cache --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:latest ./ci/docker/base
+    - docker build --build-arg CI_DOCKER_PROXY="gitlab.com/satoshilabs/dependency_proxy/containers/" --no-cache --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:latest ./ci/docker/base
```

I'd expect we'd use an environment variable to obtain the proxy address/prefix instead of hardcoding into the yamls. I.e.
```suggestion
    - docker build --build-arg CI_DOCKER_PROXY="$CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX" --no-cache --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:latest ./ci/docker/base
```

---

This does not work since the prefix only works for GitLab yml files, and as this is being done by docker from the Docker file, we need to authenticate it directly to that directory so it knows where to look. Anyway, that was the first thing I did, but docker was throwing auth errors.

---

@vdovhanych I am not sure you did this? Please do not resolve comments, respond with "Done" or commit hash as described in [firmware docs](https://docs.trezor.io/trezor-firmware/misc/review.html) so we can review your fixup.

---

Discuss in real life. The problem is that this would not work locally on your machine. You can't use a variable that you do not have obviously :). So this seems to be resolved to me!

---

> This does not work since the prefix only works for GitLab yml files

This is weird. The `CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX` variable which holds the desired value is a [CI predefined variable](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html) and hence should be available in all CI jobs, including this line.

> The problem is that this would not work locally on your machine. You can't use a variable that you do not have obviously

Note we're in a context of a *CI job* which is never run locally so this shouldn't be a problem.

What you use locally are the Dockerfiles and docker-compose files we [extended](https://github.com/satoshilabs/devops/issues/26#issuecomment-842143798) by a fallback mechanism exactly for this reason.

@vdovhanych can you maybe link a job where you tried using the variable and failed?

---

Yes, the image holds the value but only for the job in which the runner is logged in automatically, and it's pulling the image directly with the `image` option. You can take a look [here](https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/1270138618), where it failed for me. Also, looking at the link you sent, we could maybe use this `CI_DEPENDENCY_PROXY_SERVER` and try if that is working.

---

1. That job failed likely because we didn't log-in. We need to login just as we do eg. [here](https://github.com/trezor/trezor-suite/pull/3770/files#diff-8311f8c4f75bf01d23c7a9895db9bd33b3e91a9cf6694816d0c261b2ee273ebeR5)
2. The job you linked run for [this commit](https://gitlab.com/satoshilabs/trezor/trezor-suite/-/commit/94c764a40be06aae7a8f65ae4dbe68ed8d0ffde8) which, in fact, uses the hardcoded values and not the variable.

> Also, looking at the link you sent, we could maybe use this CI_DEPENDENCY_PROXY_SERVER and try if that is working.

Yeah, definitely. I see we're using a hardcoded value for logging-in. Let's use this variable instead.

To summarize what we need to do to start using the proxy *within* CI jobs is:
- [x]  extend Dockerfiles with a build arg that is used to prefix image references in the `FROM` directives.
- [x] edit docker-compose files to pass-thru this arg.
- [ ] edit invocations of `docker` and `docker-compose` in the CI jobs to pass in the correct value obtained from the environment from the `CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX` variable.
- [ ] login to the proxy in `before-script` of every job that pushes or fetches images from Dockerhub using the `CI_DEPENDENCY_PROXY_SERVER `, `CI_DEPENDENCY_PROXY_USER`  and `CI_DEPENDENCY_PROXY_PASSWORD` variables.

Orthogonal to this is using the proxy to pull images *in which* the jobs run. This is far easier as it only requires changing the `image` field of all CI jobs. This is already done in this PR. (It would be also nice to do this in a separate commit. .) )

---

@goodhoko Please take a look at it now. I think this should be enough this way.

---

@vdovhanych Looks good! Thanks! Let's squash the fixup commits and merge this.